packages/remix-react/routeModules.ts

```diff
@@ -203,10 +214,11 @@ export async function loadRouteModule(
       // on dev-time errors since it's a vite compilation error and a reload is
       // just going to fail with the same issue.  Let the UI bubble to the error
       // boundary and let them see the error in the overlay or the dev server log
-      console.error(`Error loading route module \`${route.module}\`:`, error);
       throw error;
     }
+
     window.location.reload();
```

Comment:
We could also consider enhancing this behavior in the future too.  Reloading the current page isn't necessarily what we want all the time.  There's 2 usages of this method:

* prefetching - in that case there's no real reason to hard reload since the user may not click on the link, so we could just let the prefetch no-op
* navigating - ideally for the most seamless UX we'd hard-navigate to the _next_ url instead of reloading the current URL in response to a clicked link.  The issue here is that currently `route.lazy` (which calls this) isn't aware of the next location so we'd have to plumb that through in RR if we wanted to smooth out that flow.

Reply:
It has bugged me many times that the current page gets reloaded instead of the target URL.